### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**
[DEMO LINK](https://Zelinskyi-Serhii.github.io/react_static-list-of-posts/)

### CODE REVIEW

**Review by Anton-Tretiak — APPROVED** [review]
Good Job!

File: src/components/PostInfo/PostInfo.tsx [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      <hr />
+      {comments.length === 0
+        ? (<b data-cy="NoCommentsMessage">No comments yet</b>)
+        : (<CommentList comments={comments} />
+        )}
```
Comment (illiarerun) [inline src/components/PostInfo/PostInfo.tsx:40]:
```suggestion
      {comments.length === 0
        ? (<b data-cy="NoCommentsMessage">No comments yet</b>)
        : (<CommentList comments={comments} />)
      }
```

File: src/components/PostList/PostList.tsx [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
-  <>Put the list here</>
-);
+type Props = {
+  posts: Posts[]
+};
+
+export const PostList: React.FC<Props> = ({ posts }) => {
+  return (
+    <div className="PostList">
+      {posts.map((post) => <PostInfo post={post} key={post.id} />)}
```
Comment (illiarerun) [inline src/components/PostList/PostList.tsx:12]: Again, don't write code in one line

File: src/components/PostInfo/PostInfo.tsx [hunk 3]
```diff
@@ -1,5 +1,43 @@
...
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{title}</h3>
+
+        <p>
+          {' Posted by  '}
+
+          {user && <UserInfo user={user} key={user.id} />}
```
Comment (illiarerun) [inline src/components/PostInfo/PostInfo.tsx:28]: you don't need key here, you're not mapping users
```suggestion
          {user && <UserInfo user={user} />}
```

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -1,109 +1,37 @@
...
-        <div className="PostInfo__header">
-          <h3 className="PostInfo__title">
-            doloremque illum aliquid sunt
-          </h3>
+  return foundUser || null;
+}
 
-          <p>
-            {' Posted by  '}
+function getCommentsById(postId: number): Comments[] {
```
Comment (sergii-nosachenko) [inline src/App.tsx:18]:
```suggestion
function getCommentsByPostId(postId: number): Comments[] {
```

File: src/App.tsx [hunk 5]
```diff
@@ -1,109 +1,37 @@
...
+  const foundComments = commentsFromServer
```
Comment (sergii-nosachenko) [inline src/App.tsx:19]: Immediately return

File: src/types/Comments.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export interface Comments {
```
Comment (sergii-nosachenko) [inline src/types/Comments.ts:1]:
```suggestion
export interface Comment {
```

File: src/types/Posts.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { Comments } from './Comments';
+import { Users } from './Users';
+
+export interface Posts {
```
Comment (sergii-nosachenko) [inline src/types/Posts.ts:4]:
```suggestion
export interface Post {
```

You describe one entity here

File: src/types/Users.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export interface Users {
```
Comment (sergii-nosachenko) [inline src/types/Users.ts:1]: Same here

File: src/components/CommentInfo/CommentInfo.tsx [hunk 9]
```diff
@@ -1,5 +1,35 @@
...
+  const {
+    name,
+    email,
+    body,
+  } = comment;
+
+  return (
+    <div className="CommentInfo">
+      <div className="CommentInfo__title">
+        <strong className="CommentInfo__name">{name}</strong>
```
Comment (sergii-nosachenko) [inline src/components/CommentInfo/CommentInfo.tsx:18]: Split into rows

File: src/components/CommentList/CommentList.tsx [hunk 10]
```diff
@@ -1,5 +1,20 @@
...
+
+import { Comments } from '../../types/Comments';
+import { CommentInfo } from '../CommentInfo';
+
+type Props = {
+  comments: Comments[]
+};
+
+export const CommentList: React.FC<Props> = ({ comments }) => {
+  return (
```
Comment (sergii-nosachenko) [inline src/components/CommentList/CommentList.tsx:13]: Useless `return`

File: src/components/PostInfo/PostInfo.tsx [hunk 11]
```diff
@@ -1,5 +1,42 @@
...
+    title,
+    body,
+    user,
+    comments,
+  } = post;
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{title}</h3>
```
Comment (sergii-nosachenko) [inline src/components/PostInfo/PostInfo.tsx:23]: Split

File: src/components/PostInfo/PostInfo.tsx [hunk 12]
```diff
@@ -1,5 +1,42 @@
...
+
+        <p>
```
Comment (sergii-nosachenko) [inline src/components/PostInfo/PostInfo.tsx:25]: Check if user exists here

File: src/components/PostInfo/PostInfo.tsx [hunk 13]
```diff
@@ -1,5 +1,42 @@
...
+          {user && <UserInfo user={user} />}
+        </p>
+      </div>
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      <hr />
+      {comments.length
```
Comment (sergii-nosachenko) [inline src/components/PostInfo/PostInfo.tsx:37]: Add space

File: src/components/PostList/PostList.tsx [hunk 14]
```diff
@@ -1,5 +1,20 @@
...
 
-export const PostList: React.FC = () => (
-  <>Put the list here</>
-);
+type Props = {
+  posts: Posts[]
+};
+
+export const PostList: React.FC<Props> = ({ posts }) => {
+  return (
```
Comment (sergii-nosachenko) [inline src/components/PostList/PostList.tsx:10]: Useless

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/components/PostInfo/PostInfo.tsx [hunk 15]
```diff
@@ -23,12 +23,14 @@ export const PostInfo: React.FC<Props> = ({ post }) => {
         <h3 className="PostInfo__title">
           {title}
         </h3>
+        {user && (
```
Comment (sergii-nosachenko) [inline src/components/PostInfo/PostInfo.tsx:26]:
```suggestion

        {user && (
```

File: src/components/PostInfo/PostInfo.tsx [hunk 16]
```diff
@@ -23,12 +23,14 @@ export const PostInfo: React.FC<Props> = ({ post }) => {
...
+          <p>
+            {' Posted by  '}
 
-        <p>
-          {' Posted by  '}
+            <UserInfo user={user} />
+          </p>
```
Comment (sergii-nosachenko) [inline src/components/PostInfo/PostInfo.tsx]: Paragraph is a part of UserInfo component

**Review by sergii-nosachenko — APPROVED** [review]
Good work 👍
Commented

**Pull request closed without merging**